substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/Target_jdk_jfr_internal_Logger.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2021, 2021, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2021, 2021, Red Hat Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.jfr;
+
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+import jdk.jfr.internal.LogTag;
+import jdk.jfr.internal.LogLevel;
+import jdk.jfr.internal.Logger;
+
+@TargetClass(value = jdk.jfr.internal.Logger.class, onlyWith = JfrEnabled.class)
+public final class Target_jdk_jfr_internal_Logger {
+    /** See {@link Logger#shouldLog}. */
+    @SuppressWarnings({"unused"})
+    @Substitute
+    public static boolean shouldLog(LogTag tag, LogLevel level) {
```

Comment:
The method will always return false, since it relies on hotspot to set `tagSetLevel` to the proper value. And that means the `JVM.log` call is never made.

Logger.java:
```
public static boolean shouldLog(LogTag tag, LogLevel level) {
    return level.level >= tag.tagSetLevel;
}
```
LogTag.java:
```
/* set from native side */
volatile int tagSetLevel = 100; // prevent logging if JVM log system has not been initialized
```

Reply:
Yes, I think that makes more sense than the substitution.